src/parser/constructs.rs

```diff
@@ -356,8 +364,18 @@ impl Construct {
         let (input, _) = Token::maybe_consume_extra(input)?;
 
         let (input, (instructions, last)) = Construct::stmts_and_maybe_last(input)?;
-
         let (input, _) = Token::maybe_consume_extra(input)?;
+
+        if last.is_some() {
+            let (_, dead_code) = opt(Construct::instruction)(input)?;
+            if dead_code.is_some() {
+                return Err(NomError(
+                    Error::new(ErrKind::Parsing)
+                        .with_msg(format!("Dead code after early return: {}", input)),
```

Comment:
Could you add a FIXME here stipulating that this should get changed to a warning?

Reply:
Also: This will print the `input` in the error message, while we should probably be printing the instruction instead? Maybe? I'm not sure.

But you could change it to this:
```suggestion
        if let (_, Some(dead_code)) = opt(Construct::instruction)(input)? {
                return Err(NomError(
                    Error::new(ErrKind::Parsing)
                        .with_msg(format!("dead code after early return: {}", dead_code.print())),
```